src/python/pants/backend/shell/target_types.py

```diff
@@ -368,36 +306,8 @@ class ShellCommandSourcesField(MultipleSourcesField):
     expected_num_files = 0
 
 
-class RunInSandboxSourcesField(MultipleSourcesField):
-    # We solely register this field for codegen to work.
-    alias = "_sources"
-    uses_source_roots = False
-    expected_num_files = 0
-
-
-class RunInSandboxArgumentsField(StringSequenceField):
-    alias = "args"
-    default = ()
-    help = f"Extra arguments to pass into the `{RunInSandboxRunnableField.alias}` field."
-
-
-class RunInSandboxStdoutFilenameField(StringField):
-    alias = "stdout"
-    default = None
-    help = "A filename to capture the contents of `stdout` to, relative to the value of `workdir`."
-
-
-class RunInSandboxStderrFilenameField(StringField):
-    alias = "stderr"
-    default = None
-    help = "A filename to capture the contents of `stdout` to, relative to the value of `workdir`."
-
-
-class ShellCommandTimeoutField(IntField):
-    alias = "timeout"
-    default = 30
-    help = "Command execution timeout (in seconds)."
-    valid_numbers = ValidNumbers.positive_only
+class ShellCommandTimeoutField(AdhocToolTimeoutField):
+    pass
 
 
 class ShellCommandToolsField(StringSequenceField):
```

Comment:
This should probably move to the adhoc backend as `AdhocToolToolsField` and then this should extend it.

Reply:
probably not (happy to discuss use cases)

Reply:
You've already addressed this in the other comment.